airflow/providers/amazon/aws/hooks/sagemaker.py

```diff
@@ -155,6 +155,7 @@ class SageMakerHook(AwsBaseHook):
     endpoint_non_terminal_states = {"Creating", "Updating", "SystemUpdating", "RollingBack", "Deleting"}
     pipeline_non_terminal_states = {"Executing", "Stopping"}
     failed_states = {"Failed"}
+    training_failed_states = {*failed_states, "Stopped"}
```

Comment:
Sagemaker has so many endpoints, and more popping up so often that I am not willing to guarantee that none of them do or will ever use "Stopped" as a successful end state, so I am only adding it to the Training failed states.

Reply:
"Stopped" is really a failed state? What is the state called when the training job completes successfully? Is it not Stopped?

Reply:
`'TrainingJobStatus': 'InProgress'|'Completed'|'Failed'|'Stopping'|'Stopped',
`

"Completed" is the terminal successful state.

Reply:
Got it, thanks!